File: python/백준/자료구조/스택, 집합/9935 문자열 폭발.py
string = input()
bomb = input()

stack = []
bomb_len = len(bomb)

for ch in string:
    stack.append(ch)
    # 폭탄 문자열 길이만큼 스택 끝부분이 폭탄이랑 같으면 제거
    if ''.join(stack[-bomb_len:]) == bomb:
        for _ in range(bomb_len):
            stack.pop()

result = ''.join(stack)
print(result if result else "FRULA")

File: cpp/백준/자료구조/스택, 집합/9935 문자열 폭발.cpp
#include <iostream>
#include <string>
#include <vector>
using namespace std;

int main() {
    ios_base::sync_with_stdio(false);
    cin.tie(nullptr);

    string s, bomb;
    cin >> s >> bomb;

    int bomb_len = (int)bomb.size();
    vector<char> stack;
    stack.reserve(s.size());

    for (char ch : s) {
        stack.push_back(ch);
        if ((int)stack.size() >= bomb_len) {
            bool match = true;
            for (int i = 0; i < bomb_len; ++i) {
                if (stack[stack.size() - bomb_len + i] != bomb[i]) {
                    match = false;
                    break;
                }
            }
            if (match) {
                for (int i = 0; i < bomb_len; ++i) stack.pop_back();
            }
        }
    }

    if (stack.empty()) {
        cout << "FRULA\n";
    } else {
        for (char c : stack) cout << c;
        cout << "\n";
    }
    return 0;
}
